apps/examples/testcase/le_tc/network/itc_net_setsockopt.c

```diff
@@ -0,0 +1,237 @@
+* @precondition         :socket file descriptor
+* @postcondition        :None
+*/
+static void itc_net_setsockopt_p_reuseaddr(int s)
+{
+	int ret = -1;
+	int option = 1;
+
+	ret = setsockopt(s, SOL_SOCKET, SO_REUSEADDR, (char *)&option, sizeof option);
+	TC_ASSERT_EQ("setsockopt", ret, 0);
+	TC_SUCCESS_RESULT();
+}
+
+/**
+* @testcase             :itc_net_setsockopt_p_rcvtimo
+* @brief                :setsockopt setting flag SO_RCVTIMEO
+* @scenario             :setsockopt setting flag SO_RCVTIMEO
+* @apicovered           :setsockopt()
+* @precondition         :socket file descriptor
+* @postcondition        :None
+*/
+static void itc_net_setsockopt_p_rcvtimo(int s)
+{
+	int ret = -1;
+	struct timeval tv;
+	tv.tv_sec = 1;
+	tv.tv_usec = 0;
+
+	ret = setsockopt(s, SOL_SOCKET, SO_RCVTIMEO, (struct timeval *)&tv, sizeof(struct timeval));
+
+	TC_ASSERT_EQ("setsockopt", ret, 0);
+	TC_SUCCESS_RESULT();
+}
+
+/**
+* @fn                   :itc_net_setsockopt_p_sndtimo
+* @brief                :setsockopt setting flag SO_SNDTIMEO
+* @Scenario             :setsockopt setting flag SO_SNDTIMEO
+* @API's covered        :setsockopt()
+* Preconditions         :socket file descriptor
+* Postconditions        :None
+*/
+static void itc_net_setsockopt_p_sndtimo(int s)
+{
+	int ret = -1;
+	struct timeval timeout;
+	timeout.tv_sec = 10;
+	timeout.tv_usec = 0;
+
+	ret = setsockopt(s, SOL_SOCKET, SO_SNDTIMEO, (char *)&timeout, sizeof(timeout));
+	TC_ASSERT_EQ("setsockopt", ret, 0);
+	TC_SUCCESS_RESULT();
+}
+
+/**
+* @fn                   :itc_net_setsockopt_p_no_check
+* @brief                :setsockopt setting flag SO_NO_CHECK
+* @Scenario             :setsockopt setting flag SO_NO_CHECK
+* @API's covered        :setsockopt()
+* Preconditions         :socket file descriptor
+* Postconditions        :None
+*/
+static void itc_net_setsockopt_p_no_check(int s)
+{
+	int ret = -1;
+	int optval = 1;
+
+	ret = setsockopt(s, SOL_SOCKET, SO_NO_CHECK, &optval, sizeof(optval));
+	TC_ASSERT_EQ("setsockopt", ret, 0);
+	TC_SUCCESS_RESULT();
+}
+
+/**
+* @fn                   :itc_net_setsockopt_p_ip_ttl
+* @brief                :setsockopt setting flag IP_TTL
+* @Scenario             :setsockopt setting flag IP_TTL
+* @API's covered        :setsockopt()
+* Preconditions         :socket file descriptor
+* Postconditions        :None
+*/
+static void itc_net_setsockopt_p_ip_ttl(int s)
+{
+	int ret = -1;
+	int optval = 1;
+
+	ret = setsockopt(s, IPPROTO_IP, IP_TTL, &optval, sizeof(optval));
+	TC_ASSERT_EQ("setsockopt", ret, 0);
+	TC_SUCCESS_RESULT();
+}
+
+/**
+* @fn                   :itc_net_setsockopt_p_multicast_ttl_loop
+* @brief                :setsockopt setting flag IP_MULTICAST_LOOP
+* @Scenario             :setsockopt setting flag IP_MULTICAST_LOOP
+* @API's covered        :setsockopt()
+* Preconditions         :socket file descriptor
+* Postconditions        :None
+*/
+
+static void itc_net_setsockopt_p_multicast_ttl_loop(int s)
+{
+	int ret = -1;
+	u8_t loop = 250;
```

Comment:
Why do you use ```u8_t```? Anyway you need to include header for that.

Reply:
changes are done as per your suggestion. please review